src/realm/sync/noinst/client_impl_base.cpp

```diff
@@ -1284,6 +1266,31 @@ void Connection::receive_error_message(int raw_error_code, StringData message, b
 }
 
 
+void Connection::receive_query_error_message(int raw_error_code, std::string_view message, int64_t query_version,
+                                             session_ident_type session_ident)
+{
+    if (session_ident == 0) {
+        logger.error("Received query error message for session ident 0."); // throws;
+        return close_due_to_protocol_error(ClientError::bad_session_ident);
+    }
+
+    if (!is_flx_sync_connection()) {
+        logger.error("Received query error message on a non-FLX sync connection");
+        return close_due_to_protocol_error(ClientError::bad_protocol_from_server);
+    }
+
+    auto session = get_session(session_ident);
+    if (!session) {
+        logger.error("Bad session identifier in QUERY_ERROR mesage, session_ident = %1", session_ident); // throws
+        return close_due_to_protocol_error(ClientError::bad_session_ident);                              // throws
+    }
+
+    if (auto ec = session->receive_query_error_message(raw_error_code, message, query_version); ec) {
```

Comment:
If this is a style thing then feel free to ignore, but I'd opt for simplicity:
```suggestion
    if (auto ec = session->receive_query_error_message(raw_error_code, message, query_version)) {
```

Reply:
done!